visual_qa/florence-2/main.py

```diff
@@ -89,18 +92,145 @@ def __predict__(
+                raise Exception("Failed to open video file.")
+
+            if start_frame > 0:
+                cap.set(cv2.CAP_PROP_POS_FRAMES, start_frame)
+
+            width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+            height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+            fps = cap.get(cv2.CAP_PROP_FPS) / frame_interval
+
+            if end_frame > 0:
+                end_frame = min(end_frame, int(cap.get(cv2.CAP_PROP_FRAME_COUNT)))
+
+            if debug_visualization:
+                tmp_out_path = "output.mp4"
+                if os.path.exists(tmp_out_path):
+                    os.remove(tmp_out_path)
+                out_vid_writer = cv2.VideoWriter(tmp_out_path, cv2.VideoWriter_fourcc(*"mp4v"), fps, (width, height))
+
+            frame_count = 0
+            approx_frame_total = int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
+            parsed_answer_list = []
+            while True:
+                if frame_count // frame_interval % 100 == 0:
+                    print(f"Processing frame {frame_count}/{approx_frame_total}...")
+                ret, frame = cap.read()
+                if not ret or (end_frame > 0 and frame_count > end_frame):
+                    break
+
+                out = self.process_image(frame, task_prompt, text_input, debug_visualization)
+                if debug_visualization:
+                    out_image, parsed_answer = out
+                    out_vid_writer.write(out_image)
+                else:
+                    parsed_answer = out
+
+                parsed_answer["frame_number"] = frame_count
+
+                parsed_answer_list.append(parsed_answer)
+
+                frame_count += 1
+
+                for _ in range(frame_interval - 1):
+                    ret, _ = cap.read()
+                    frame_count += 1
+                    if not ret or (end_frame > 0 and frame_count > end_frame):
+                        break 
+                
 
+            cap.release()
+                
+
+            if debug_visualization:
+                out_vid_writer.release()
+                import subprocess
+
+                if os.path.exists("final_output.mp4"):
+                    os.remove("final_output.mp4")
+
+                print("Re-encoding the video with the original audio track...")
+
+                # Use ffmpeg to re-encode the video and add the original audio track
+                command = [
+                    "ffmpeg",
+                    "-loglevel", "error", # Suppress ffmpeg logs
+                    "-i", tmp_out_path,  # Input the processed video without audio
+                    "-i", file_path,  # Input the original video with audio
+                    "-y",  # Overwrite the output file if it already exists
+                    "-c:v", "libx264",  # Re-encode the video using H.264 codec
+                    "-map", "0:v:0",  # Map the video from the processed video
+                    "-map", "1:a:0",  # Map the audio from the original video
```

Comment:
use "-map 1:a:0?"

this will handle videos without and audio stream without crashing

Reply:
great catch ty